chain/transaction.go

```diff
@@ -174,6 +154,31 @@ type Transaction struct {
 	stateKeys state.Keys
 }
 
+// newTransaction creates a Transaction and initializes the private fields.
+func newTransaction(base *Base, actions Actions, auth Auth) (*Transaction, error) {
+	tx := Transaction{
+		TransactionData: TransactionData{
+			Base:    base,
+			Actions: actions,
+		},
+		Auth: auth,
+	}
+	p := codec.NewWriter(0, consts.NetworkSizeLimit)
+	if _, err := tx.UnsignedBytes(); err != nil {
+		return nil, err
+	}
+	if err := tx.Marshal(p); err != nil {
+		return nil, err
+	}
+	if err := p.Err(); err != nil {
+		return nil, err
+	}
```

Comment:
Could we move creating the codec to directly before it's used and use the unsigned bytes to calculate how much memory to pre-allocate for the transaction (same as before this should be `len(unsignedBytes) + consts.ByteLen + auth.Size)` )